### DISCUSSION TITLE
backref still OK for adjacency list relationship?

### DISCUSSION THREAD

**Discussion by ianjosephwilson** [sqlalchemy/sqlalchemy#9011] [Usage Questions]
The docs for adjacency relationships still use `backref` whereas in the rest of the docs including the docs for `backref` itself there are notes about `backref` being "legacy".  

Should the docs be using two relationships here with `back_populates` in the same class?  

The docs specifically here [adjacency-list-relationships](https://docs.sqlalchemy.org/en/20/orm/self_referential.html#adjacency-list-relationships) start with a children only relationship and then add a parent only relationship and then add the use of `backref` to supply both.  Should this just be both relationships with the addition of `back_populates`?  Seems kind of inconsistent but saves some complexity in the definitions.  I thought `backref` was on its way out so I was surprised to see it here.  I've been answering questions on stackoverflow off and on and this came up and I'm not sure what the best recommendation is.

```
class Node(Base):
    __tablename__ = 'nodes'

    id = Column(Integer, primary_key=True)
    name= Column(String)

    parent_id = Column(Integer, ForeignKey('nodes.id'), nullable=True)

    # children = relationship("User", backref=backref("parent", remote_side=id))
    parent = relationship('Node', back_populates='children', remote_side=id)
    children = relationship("Node", back_populates='parent', remote_side=parent_id)
```

**edit** fixed `remote_side` in `children` comment

**Comment by CaselIT** [comment]
Hi,

These docs are not yet updated. They are also not using v2 style orm classes yet.
In general everything still works the same wrt backref even where it's no longer used in the docs, but back_populates should be preferred, expecially when combined with typing tools
